### PR TITLE
Add async-to-promises Babel plugin

### PR DESCRIPTION
**Pull request by lencioni**
This plugin claims to enable async/await without generators.

https://github.com/marten-de-vries/kneden

### CODE REVIEW

**Comment by goatslacker** [comment]
We should look out for these cases https://github.com/marten-de-vries/kneden/issues/13

File: package.json [hunk 1]
```diff
@@ -16,11 +16,12 @@
   ],
   "license": "Apache-2.0",
   "dependencies": {
+    "babel-plugin-async-to-promises": "^1.0.4",
     "babel-plugin-transform-es2015-template-literals": "^6.5.0",
     "babel-plugin-transform-es3-member-expression-literals": "^6.5.0",
     "babel-plugin-transform-es3-property-literals": "^6.5.0",
-    "babel-plugin-transform-jscript": "^6.5.0",
```
Comment (ljharb) [inline package.json]: iirc the order of these is important - they should not be alphabetized.

Comment (lencioni) [inline package.json]: This was done automatically by `npm install`. Why would the order of the dependencies in the package.json file affect anything?

Comment (ljharb) [inline package.json]: Oh nvm, I though this was the Babel config. Don't mind me :-)

**Comment by ljharb** [comment]
The change looks good, but I don't think we're ready to put this transform in the preset until we've tested it out a bit and it handles a few more cases.

**Comment by lencioni** [comment]
Cool cool cool, makes sense.

**Comment by tleunen** [comment]
Just by curiosity, what do you think of [fast-async](https://github.com/MatAtBread/fast-async)? Would this be a good candidate too?

**Comment by ljharb** [comment]
To clarify - for us to be able to use async/await, we need a transform (any transform) that either a) transforms 100% of the spec, or b) can be combined with a linter rule that fully forbids the parts of the spec that it can't transpile.

**Comment by lencioni** [comment]
I don't plan on working on this any time soon. Some brave soul should feel free to resurrect.

**Pull request closed without merging**